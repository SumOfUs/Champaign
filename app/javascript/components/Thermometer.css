.Thermometer-stats {
  width: 100%;
  line-height: 14px;
  font-size: 0.9em;
  font-weight: bold;
  text-transform: uppercase;
}
.Thermometer-temperature {
  width: auto;
  float: left;
  text-align: left;
}
.Thermometer-goal {
  width: auto;
  text-align: right;
  color: rgba(48, 57, 79, 0.5);
}

.Thermometer-bg {
  background-color: white;
  border-color: #aaa;
  border-radius: 4px;
  clear: both;
  height: 16px;
  margin: 5px 0 20px;
  padding: 0;
  position: relative;
  width: 100%;
}

.Thermometer-mercury {
  border-radius: 4px;
  background-color: #00c0cf;
  margin: 0;
  padding: 0;
  height: 100%;
  min-width: 8px;
}
