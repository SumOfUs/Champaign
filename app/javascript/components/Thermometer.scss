
.Thermometer {

}

.Thermometer-title {
  font-weight: bold;
  margin-bottom: 1em;
}

.Thermometer-stats {
  width: 100%;
  line-height: 14px;
  font-size: 12px;
  font-weight: bold;
  text-transform: uppercase;
}
.Thermometer-temperature {
  width: auto;
  float: left;
  text-align: left;
}
.Thermometer-goal {
  width: auto;
  text-align: right;
  color: rgba(48, 57, 79, 0.5);
}

.Thermometer-bg {
  background-color: white;
  border-color: #aaa;
  border-radius: 4px;
  clear: both;
  height: 16px;
  margin: 5px 0 20px;
  padding: 0;
  position: relative;
  width: 100%;
}

.Thermometer-mercury {
  border-radius: 4px;
  background-color: #00c0cf;
  margin: 0;
  padding: 0;
  height: 100%;
  min-width: 8px;
}

/* Free standing version */
.fundraiser-bar--freestanding {
  .Thermometer-bg {
    background-color: #f5f5f5;
  }
}
