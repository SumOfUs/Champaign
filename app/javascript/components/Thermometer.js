// @flow
import React from 'react';
import { connect } from 'react-redux';
import { isEmpty, min } from 'lodash';
import CurrencyAmount from './CurrencyAmount';
import './Thermometer.css';
import type { AppState } from '../state/reducers';

type Props =
  | {}
  | {
      active: boolean,
      currency: string,
      donations: number,
      goal: number,
      offset: number,
      remaining: number,
      title: ?string,
    };

export function Thermometer(props: Props) {
  // Only render if active
  if (isEmpty(props) || !props.active) return null;

  // Prevent overflow when donations > goal.
  const donations = min([props.donations, props.goal]);

  return (
    <div className="Thermometer">
      <p className="Thermometer-title">{props.title}</p>
      <div className="Thermometer-stats">
        <div className="Thermometer-temperature">
          <CurrencyAmount amount={donations} currency={props.currency} />
        </div>
        <div className="Thermometer-goal">
          <CurrencyAmount amount={props.goal} currency={props.currency} />
        </div>
      </div>
      <div className="Thermometer-bg">
        <div
          className="Thermometer-mercury"
          style={{
            width: `${Math.round((donations / props.goal) * 100)}%`,
          }}
        />
      </div>
    </div>
  );
}

const mapStateToProps = (state: AppState): Props => {
  const data = state.donationsThermometer;
  const currency = state.fundraiser.currency;
  if (isEmpty(data)) return {};
  return {
    active: data.active,
    currency,
    donations: data.totalDonations[currency],
    goal: data.goals[currency],
    offset: data.offset,
    title: data.title,
  };
};

export default connect(mapStateToProps)(Thermometer);
