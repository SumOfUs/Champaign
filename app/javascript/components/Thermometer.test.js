import React from 'react';
import renderer from 'react-test-renderer';
import { IntlProvider } from 'react-intl';
import { Thermometer } from './Thermometer';

test('Renders as expected', () => {
  const component = renderer.create(
    <IntlProvider locale="en-GB" messages={{}}>
      <Thermometer
        title="Title"
        donations={1000}
        goal={600000}
        currency={'GBP'}
      />
    </IntlProvider>
  );

  expect(component.toJSON()).toMatchSnapshot();
});
