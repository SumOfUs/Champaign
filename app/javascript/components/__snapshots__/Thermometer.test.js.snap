// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders as expected 1`] = `
<div
  className="Thermometer"
>
  <div
    className="Thermometer-stats"
  >
    <div
      className="Thermometer-temperature"
    >
      <span>
        £1,000
      </span>
    </div>
    <div
      className="Thermometer-goal"
    >
      <span>
        £600,000
      </span>
    </div>
  </div>
  <div
    className="Thermometer-bg"
  >
    <div
      className="Thermometer-mercury"
      style={
        Object {
          "backgroundColor": "#00c0cf",
          "width": "0%",
        }
      }
    />
  </div>
</div>
`;
